app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -157,26 +178,58 @@ def areNetworked(
     def removeXCoal(self, X: int, towns: List[Town], player: Player):
         for town in towns:
             availableCoal = self.getAvailableCoalBuildingsTradePosts(town)
-            if availableCoal == 0:
+            if len(availableCoal) == 0:
                 continue
 
             _available = availableCoal.pop(
                 0
-            )  # todo assert these changes to resourceAmount are actually affecting the building
-            # take coal away from resources
+            )
             while X > 0:
                 if _available.type == "TradePost":
                     cost = self.priceForCoal(X)
-                    player.money -= cost  # todo assert this changes player's money
+                    player.pay(cost)
                     self.coalMarketRemaining -= X
```

Comment:
Not part of this MR, but similar to my iron comment above, we should make sure this doesn't go below 0

Reply:
It does that in the `pay` function

Reply:
No, the ‘coalMarketRemaining’ shouldn’t go below 0

Reply:
Added